Ocamlmklib reverses the order of my -L options.

<!--
Welcome to OCaml's Issue tracker!

OCaml's developers use this tracker for bugs and feature requests only, rather
than user support.

If you have questions about *using* OCaml, please ask at
https://discuss.ocaml.org (more people read Discuss than this tracker, and
you'll get confirmation of whether you've really found a bug or need a new
feature).

If your error came from the OCaml package manager, opam, (messages beginning
`[ERROR] The compilation of ...`), please start at
https://github.com/ocaml/opam-repository/issues/new.

Some libraries and tools which used to be part of OCaml are now maintained
separately. Please post questions about Graphics, Num, camlp4, LablTk, CamlDBM
or OCamlTopWin on Discuss or on their respective issue trackers (see [README.adoc](https://github.com/ocaml/ocaml/blob/trunk/README.adoc#separately-maintained-components)
for a full list).
-->

### Description

The `ocamlmklib` tool incorrectly reverses the order of `-L` library path options, which breaks the intended library search precedence and can cause linking failures or incorrect library selection.

### Example

`echo "int dummy = 0;" > dummy.c && gcc -c dummy.c`
`ocamlmklib -verbose -L/custom/lib -L/usr/lib -L/opt/lib -ltest -o test dummy.o`

Output:

`> + cc -shared -flat_namespace -undefined suppress                    -Wl,-no_compact_unwind  -o ./dlltest.so dummy.o    -L/opt/lib -L/usr/lib -L/custom/lib -ltest `

--- a/Changes
+++ b/Changes
@@ -247,6 +247,8 @@ Working version
   fix all occurences
   (Stefan Muenzel, review by Gabriel Scherer)
 
+- #14172 : Fix ocamlmklib ordering of -L options
+  (Alexandre Douard, )
 
 OCaml 5.4.0
 ---------------

--- a/Changes
+++ b/Changes
@@ -406,7 +406,7 @@ Working version
   of nonmatching input if a rule is not exhaustive (Martin Jambon,
   review by David Allsopp and Gabriel Scherer)
 
-- #14172: stop reversing the order of -L arguments in ocamlmklib, a quirk
+* #14172: stop reversing the order of -L arguments in ocamlmklib, a quirk
   introduced by accident in 2012.
   (Alexandre Douard, review by Daniel Bünzli and Nicolás Ojeda Bär)
 

--- a/Changes
+++ b/Changes
@@ -406,6 +406,10 @@ Working version
   of nonmatching input if a rule is not exhaustive (Martin Jambon,
   review by David Allsopp and Gabriel Scherer)
 
+- #14172: stop reversing the order of -L arguments in ocamlmklib, a quirk
+  introduced by accident in 2012.
+  (Alexandre Douard, review by Daniel Bünzli and Nicolás Ojeda Bär)
+
 ### Manual and documentation:
 
 - #14598: Add `@since` annotations to `Runtime_events`. (Raphaël Proust)

--- a/tools/ocamlmklib.ml
+++ b/tools/ocamlmklib.ml
@@ -172,7 +172,7 @@ let parse_arguments argv =
   List.iter
     (fun r -> r := List.rev !r)
     [ bytecode_objs; native_objs; caml_libs; caml_opts;
-      c_libs; c_objs; c_opts; ld_opts; rpath ];
+      c_libs; c_objs; c_opts; ld_opts; rpath; c_Lopts ];
 (* Put -L options in front of -l options in -cclib to mimic -ccopt behavior *)
   c_libs := !c_Lopts @ !c_libs;
 

--- a/tools/ocamlmklib.ml
+++ b/tools/ocamlmklib.ml
@@ -175,7 +175,7 @@ let parse_arguments argv =
   List.iter
     (fun r -> r := List.rev !r)
     [ bytecode_objs; native_objs; caml_libs; caml_opts;
-      c_libs; c_objs; c_opts; ld_opts; rpath ];
+      c_libs; c_objs; c_opts; ld_opts; rpath; c_Lopts ];
 (* Put -L options in front of -l options in -cclib to mimic -ccopt behavior *)
   c_libs := !c_Lopts @ !c_libs;
 

